group-coordinator/src/main/java/org/apache/kafka/coordinator/group/generic/GenericGroup.java

```diff
@@ -561,7 +625,7 @@ public boolean hasReceivedSyncFromAllMembers() {
      * @return members that have yet to sync.
      */
     public Set<String> allPendingSyncMembers() {
-        return pendingSyncMembers;
+        return new HashSet<>(pendingSyncMembers);
```

Comment:
Why do we need a copy here?

Reply:
in `GroupMetadataManager#expirePendingSync()` we remove members from the set while iterating

Reply:
Would it make sense to make the copy on the other side? It is a bit weird to anticipate this here because we don't do this for other accessors. I am usually tempted to return unmodifiable collections in the case to prevent this kind of issue.

Reply:
This was not addressed.

Reply:
thought i addressed this. addressed it now